daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala

```diff
@@ -37,7 +37,8 @@ private[validation] object Serializability {
 
     def checkType(): Unit = checkType(typeToSerialize)
 
-    def checkType(typ0: Type): Unit = typ0 match {
+    def checkType(typ0: Type): Unit = typ0 match { //TODO: fix the exhaustiveness checking
+      case TSynApp(syn,_) => unserializable(URDataType(syn))
```

Comment:
In the Haskell typechecker you throw `URTypeSyn` instead of `URDataType`. I think we should be consistent here.

Reply:
Thanks, nice catch. I wrote the `URTypeSyn` case class, but forgot to use it.

Reply:
fixed now